src/main/java/net/barrage/school/java/ecatalog/app/productSource/ProductSource.java

```diff
@@ -14,5 +15,7 @@ interface Factory {
         ProductSource create(ProductSourceProperties.SourceProperty psp);
     }
 
-    List<Product> getProducts();
+    List<Product> getProducts(Merchant merchant);
```

Comment:
Looks odd... U have a product source. Why do u need any `Merchant` to get products from it?